aiida_vasp/io/tests/test_poscar_io.py

```diff
@@ -15,11 +15,136 @@ def test_poscar_io(fresh_aiida_env, vasp_structure_poscar):
+    structure.
+
+    """
+
+    path = data_path('poscar', 'POSCAR')
+    parser = PoscarParser(file_path=path)
+    result = parser.get_quantity('poscar-structure', {})
+    structure = vasp_structure
+    assert result['poscar-structure'].cell == structure.cell
+    assert result['poscar-structure'].get_site_kindnames() == structure.get_site_kindnames()
+    assert result['poscar-structure'].sites[2].position == structure.sites[2].position
+
+
+@pytest.mark.parametrize(['vasp_structure'], [('str-Al',)], indirect=True)
+def test_parse_poscar_write(fresh_aiida_env, vasp_structure, tmpdir):
+    """
+    Parse (write) a reference POSCAR file.
+
+    Using the PoscarParser, parse (read), and compare to reference
+    structure.
+
+    """
+
+    structure = vasp_structure
+    parser = PoscarParser(data=structure)
+    result = parser.get_quantity('poscar-structure', {})
+    assert result['poscar-structure'].cell == structure.cell
+    assert result['poscar-structure'].get_site_kindnames() == structure.get_site_kindnames()
+    assert result['poscar-structure'].sites[2].position == structure.sites[2].position
+    temp_file = str(tmpdir.join('POSCAR'))
+    parser.write(temp_file)
+    parser = PoscarParser(file_path=temp_file)
+    result_reparse = parser.get_quantity('poscar-structure', {})
+    assert result_reparse['poscar-structure'].cell == structure.cell
+    assert result_reparse['poscar-structure'].get_site_kindnames() == \
+        structure.get_site_kindnames()
+    assert result_reparse['poscar-structure'].sites[2].position == \
+        structure.sites[2].position
+
+
+@pytest.mark.parametrize(['vasp_structure'], [('str-Al',)], indirect=True)
+def test_parse_poscar_reparse(fresh_aiida_env, vasp_structure, tmpdir):
+    """
+    Parse (read) a reference POSCAR file.
+
+    Using the PoscarParser, parse(write), parse (read), and compare
+    to reference structure.
+
+    """
+
     path = data_path('poscar', 'POSCAR')
-    parser = PoscarParser(path)
-    result = parser.get_quantity('structure', {})
+    parser = PoscarParser(file_path=path)
+    _ = parser.get_quantity('poscar-structure', {})
+    temp_file = str(tmpdir.join('POSCAR'))
+    parser.write(temp_file)
+    parser = PoscarParser(file_path=temp_file)
+    result_reparse = parser.get_quantity('poscar-structure', {})
     structure = vasp_structure
-    assert result['structure'].cell == structure.cell
-    assert result['structure'].get_site_kindnames() == structure.get_site_kindnames()
-    assert result['structure'].sites[2].position == structure.sites[2].position
+    assert result_reparse['poscar-structure'].cell == structure.cell
+    assert result_reparse['poscar-structure'].get_site_kindnames() == \
+        structure.get_site_kindnames()
+    assert result_reparse['poscar-structure'].sites[2].position == \
+        structure.sites[2].position
+
+
+# The following tests can be enabled when Aiida gets updated
```

Comment:
Could you outcomment those and mark them with an xfail which depends on aiida.__version__ >= 1.0? Because we know the feature will be in there.

Reply:
Done